src/docfx/lib/HtmlUtility.cs

```diff
@@ -163,7 +163,10 @@ public static string TransformLinks(string html, Func<string, string> transform)
             return result.ToString();
         }
 
-        public static HtmlNode RemoveTitle(HtmlNode node)
+        /// <summary>
+        /// Get title node and remove title node if all previous nodes are invisible
+        /// </summary>
+        public static (HtmlNode headerNode, bool removed) ExtractTitle(HtmlNode node)
```

Comment:
consider 
 - Having 2 methods `string GetTitle` and `string ExtractRawTitle` to indicate `rawTitle` and `title` are distinct properties
--or--
 - Having 1 method that returns `(string title, string rawTitle, HtmlDom content) SplitHtml`

That might better express the intend then leaking the `removed` intermediate state to the caller

Reply:
ExtractTitle method has clear described this method will return header node and try remove header node. 

And RawTitle is just to reuse the result returned from this method, for me it's really clear.